sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -80,6 +80,12 @@ abstract class Expression extends TreeNode[Expression] {
 
   def nullable: Boolean
 
+  /**
+   * Indicates whether this expression is null intolerant. If this is true,
+   * then any null input will result in null output).
+   */
+  def nullIntolerant: Boolean = true
```

Comment:
Also, I think its better to do it with a `trait` since then you can use InteliJ to list all the instances.